cfn-resources/resource-policy/cmd/resource/resource.go

```diff
@@ -0,0 +1,185 @@
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	orgID := currentModel.OrgId
+	resourcePolicyReq := NewResourcePolicyCreateReq(currentModel)
+	resourcePolicyResp, apiResp, err := conn.AtlasResourcePoliciesApi.CreateAtlasResourcePolicy(ctx, *orgID, resourcePolicyReq).Execute()
+	if err != nil {
+		return handleError(apiResp, constants.CREATE, err)
+	}
+	resourceModel := GetResourcePolicyModel(resourcePolicyResp, currentModel)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Completed",
+		ResourceModel:   resourceModel,
+	}, nil
+}
+
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	conn, peErr := initEnvWithLatestClient(req, currentModel, ReadRequiredFields)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	orgID := currentModel.OrgId
+	resourcePolicyID := currentModel.Id
+	resourcePolicyResp, apiResp, err := conn.AtlasResourcePoliciesApi.GetAtlasResourcePolicy(ctx, *orgID, *resourcePolicyID).Execute()
+	if err != nil {
+		return handleError(apiResp, constants.READ, err)
+	}
+
+	resourceModel := GetResourcePolicyModel(resourcePolicyResp, currentModel)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModel:   resourceModel,
+	}, nil
+}
+
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	conn, peErr := initEnvWithLatestClient(req, currentModel, UpdateRequiredFields)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	orgID := currentModel.OrgId
+	resourcePolicyID := currentModel.Id
+	_, _, err := conn.AtlasResourcePoliciesApi.GetAtlasResourcePolicy(ctx, *orgID, *resourcePolicyID).Execute()
+	if err != nil {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "Update failed",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound,
+		}, nil
+	}
+	resourcePolicyReq := NewResourcePolicyUpdateReq(currentModel)
+	resourcePolicyResp, apiResp, err := conn.AtlasResourcePoliciesApi.UpdateAtlasResourcePolicy(ctx, *orgID, *resourcePolicyID, resourcePolicyReq).Execute()
+	if err != nil {
+		return handleError(apiResp, constants.UPDATE, err)
+	}
+
+	resourceModel := GetResourcePolicyModel(resourcePolicyResp, currentModel)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update Completed",
+		ResourceModel:   resourceModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	conn, peErr := initEnvWithLatestClient(req, currentModel, DeleteRequiredFields)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	ctx := context.Background()
+
+	orgID := currentModel.OrgId
+	resourcePolicyID := currentModel.Id
+	if _, apiResp, err := conn.AtlasResourcePoliciesApi.DeleteAtlasResourcePolicy(ctx, *orgID, *resourcePolicyID).Execute(); err != nil {
```

Comment:
[nit] prefer using two lines

Reply:
btw: don't remember if we should allow not found here

Reply:
changed to 2 lines.